pax-std/pax-std-primitives/src/button.rs

```diff
@@ -33,92 +34,102 @@ impl InstanceNode for ButtonInstance {
-            let stk = &expanded_node.stack;
-            handle_vtable_update(tbl, stk, &mut properties.label, context.borrow().globals());
-
-            // Style
-            handle_vtable_update(tbl, stk, &mut properties.style, context.borrow().globals());
-            let stl = properties.style.get();
-            handle_vtable_update(tbl, stk, &stl.fill, context.borrow().globals());
-            handle_vtable_update(tbl, stk, &stl.font, context.borrow().globals());
-            handle_vtable_update(tbl, stk, &stl.font_size, context.borrow().globals());
-            handle_vtable_update(tbl, stk, &stl.underline, context.borrow().globals());
-            handle_vtable_update(tbl, stk, &stl.align_vertical, context.borrow().globals());
-            handle_vtable_update(tbl, stk, &stl.align_horizontal, context.borrow().globals());
-            handle_vtable_update(tbl, stk, &stl.align_multiline, context.borrow().globals());
-        });
-    }
-
-    fn handle_native_patches(
-        &self,
-        expanded_node: &ExpandedNode,
-        context: &Rc<RefCell<RuntimeContext>>,
-    ) {
-        let id_chain = expanded_node.id_chain.clone();
-        let mut patch = ButtonPatch {
-            id_chain: id_chain.clone(),
-            ..Default::default()
-        };
-        let mut last_patches = self.last_patches.borrow_mut();
-        let old_state = last_patches.entry(id_chain.clone()).or_insert(ButtonPatch {
-            id_chain,
-            ..Default::default()
-        });
-
-        expanded_node.with_properties_unwrapped(|properties: &mut Button| {
-            let computed_tab = &expanded_node.layout_properties;
-            let (width, height) = computed_tab.bounds.get();
-            let updates = [
-                patch_if_needed(
-                    &mut old_state.content,
-                    &mut patch.content,
-                    properties.label.get().string.clone(),
-                ),
-                patch_if_needed(
-                    &mut old_state.style,
-                    &mut patch.style,
-                    (&properties.style.get()).into(),
-                ),
-                patch_if_needed(&mut old_state.size_x, &mut patch.size_x, width),
-                patch_if_needed(&mut old_state.size_y, &mut patch.size_y, height),
-                patch_if_needed(
-                    &mut old_state.transform,
-                    &mut patch.transform,
-                    computed_tab.transform.get().coeffs().to_vec(),
-                ),
-            ];
-            if updates.into_iter().any(|v| v == true) {
-                context
-                    .borrow_mut()
-                    .enqueue_native_message(pax_message::NativeMessage::ButtonUpdate(patch));
-            }
-        });
+        //trigger computation of and sending update message if needed
+        for native_message_prop in self.native_message_props.borrow().values() {
+            native_message_prop.get();
```

Comment:
small optimization but we could probably use the expanded node in this function to just call get once. This will recurse through the list on for every repeated expanded node on each expanded node.

Reply:
Ooops! Makes sense :)